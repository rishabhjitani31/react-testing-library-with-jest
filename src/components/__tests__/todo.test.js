import { render, screen, cleanup} from '@testing-library/react';
import Todo from '../todo';
import renderer from 'react-test-renderer';

afterEach(() => {
	cleanup();
})

test('should render non completed todo Component' ,() => {
	const todo = {
		id: 1,
		title: 'Make Dinner',
		completed: false
	      }
	render(<Todo todo={todo}/>);
	const todoElemnet = screen.getByTestId('todo-1');
	expect(todoElemnet).toBeInTheDocument();
	expect(todoElemnet).toHaveTextContent('Make Dinner');
})

test('should render completed todo Component' ,() => {
	const todo = {
		id: 2,
		title: 'Wash Dishes',
		completed: true
	      }
	render(<Todo todo={todo}/>);
	const todoElemnet = screen.getByTestId('todo-2');
	expect(todoElemnet).toBeInTheDocument();
	expect(todoElemnet).toHaveTextContent('Wash Dishes');
})

test('snapshot testing', () => {
	const todo = {
		id: 2,
		title: 'Wash Dishes',
		completed: true
	}
	const tree = renderer.create(<Todo todo={todo} />).toJSON()
	expect(tree).toMatchSnapshot();
})



