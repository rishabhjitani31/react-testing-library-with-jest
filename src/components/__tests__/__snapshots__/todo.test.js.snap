// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      u
      Wash Dishes
    </h1>
  </strike>
</div>
`;
